src/plugins/intel_npu/src/compiler/src/graph_transformations.cpp

```diff
@@ -34,20 +34,27 @@ IR serializeToIR(const std::shared_ptr<const ov::Model>& origModel, uint32_t sup
     // precision/layout preprocessing requirement. We are setting this value to "true" since the API version is no
     // longer a cause for altering the metadata. This is due to the preprocessing performed in the OpenVINO framework's
     // implementaion, the "ov::Model" object is preprocessed before reaching the NPU plugin.
-    const auto new_api_key = "is_new_api";
+    const auto newAPIKey = "is_new_api";
+
+    // Flag used for indicating an NPU plugin version which switched the I/O identification convention from names to
+    // indices. The flag is required in order to inform the driver-compiler adapter to expect indices when attempting to
+    // deserialize the I/O metadata.
+    const auto useIndicesForIOMetadata = "use_indices_for_io_metadata";
```

Comment:
No plugin versioning unfortunately, we have to rely on flags for signaling custom behavior inside the driver-compiler adapter.